rust-src/bulletproofs/src/inner_product_proof.rs

```diff
@@ -295,81 +297,60 @@ pub fn verify_inner_product<C: Curve>(
     Q: &C,
     proof: &InnerProductProof<C>,
 ) -> bool {
+    // call verify_inner_product_with_scalars
+    // Since H is directly given, set all exponents for H to 1
     let n = G_vec.len();
-    let L_R = &proof.lr_vec;
-    let a = proof.a;
-    let b = proof.b;
-    let mut ab = a;
-    ab.mul_assign(&b);
-
-    let verification_scalars = match verify_scalars(transcript, n, proof) {
-        None => return false,
-        Some(scalars) => scalars,
-    };
-    let (u_sq, u_inv_sq, s) = (
-        verification_scalars.u_sq,
-        verification_scalars.u_inv_sq,
-        verification_scalars.s,
-    );
-
-    let G = multiexp(G_vec, &s);
-    let mut s_inv = s;
-    s_inv.reverse();
-    let H = multiexp(H_vec, &s_inv);
-
-    let mut sum = C::zero_point();
-    for j in 0..L_R.len() {
-        sum = sum.plus_point(
-            &(L_R[j]
-                .0
-                .mul_by_scalar(&u_sq[j])
-                .plus_point(&(L_R[j].1.mul_by_scalar(&u_inv_sq[j])))),
-        );
-    }
-
-    let RHS = G
-        .mul_by_scalar(&a)
-        .plus_point(&H.mul_by_scalar(&b))
-        .plus_point(&Q.mul_by_scalar(&ab))
-        .minus_point(&sum);
-    P_prime.minus_point(&RHS).is_zero_point()
+    let H_exponents = vec![C::Scalar::one(); n];
+    // P_prime is also given directly, so set P_prime_bases to P_prime with exponent
+    // 1. Since it is assumed that P_prime_bases starts with G, H and Q, add those
+    // first and set the first 2n+1 exponents to 0.
+    let mut P_prime_bases = Vec::with_capacity(2 * n + 2);
+    P_prime_bases.extend(G_vec);
+    P_prime_bases.extend(H_vec);
```

Comment:
I think it would be better to use `extend_from_slice`. 

But it does not matter in practice. It is just more precise a method that does exactly what you need here more directly than the generic extend.